web/src/beta/components/Modal/index.tsx

```diff
@@ -1,54 +1,163 @@
-import { ReactNode } from "react";
+import { ReactNode, useState } from "react";
 
 import Wrapper from "@reearth/beta/components/Modal/ModalFrame";
-import Text from "@reearth/beta/components/Text";
-import { styled, useTheme } from "@reearth/services/theme";
+import { styled } from "@reearth/services/theme";
+
+type Size = "sm" | "md" | "lg";
+
+export type SidebarTab = {
+  id: string;
+  label: string;
+  content: ReactNode;
+  tabButton1?: ReactNode;
+  tabButton2?: ReactNode;
+};
 
 type Props = {
   className?: string;
-  title?: string;
-  size?: "sm" | "md" | "lg";
+  modalTitle?: string;
+  size?: Size;
   button1?: ReactNode;
   button2?: ReactNode;
   children?: ReactNode;
   isVisible?: boolean;
   onClose?: () => void;
+  showSidebar?: boolean;
+  sidebarTabs?: SidebarTab[];
 };
 
 const Modal: React.FC<Props> = ({
   className,
-  title,
-  size,
+  modalTitle,
+  size = "md",
   button1,
   button2,
+  children,
   isVisible,
   onClose,
-  children,
+  sidebarTabs = [],
 }) => {
-  const theme = useTheme();
+  const [selectedTab, setSelectedTab] = useState<string | null>(
+    sidebarTabs.length > 0 ? sidebarTabs[0].id : null,
+  );
+
+  const handleTabChange = (tabId: string) => {
+    setSelectedTab(tabId);
+  };
+
   return (
-    <Wrapper className={className} size={size} isVisible={isVisible} onClose={onClose}>
-      <Title size="h4" weight="bold" color={theme.general.content.strong}>
-        {title}
-      </Title>
-      {children}
-      <ButtonWrapper>
-        {button2}
-        {button1}
-      </ButtonWrapper>
+    <Wrapper
+      className={className}
+      size={size}
+      isVisible={isVisible}
+      modalTitle={modalTitle}
+      onClose={onClose}>
+      {sidebarTabs.length > 0 ? (
+        <SidebarWrapper>
+          <NavBarWrapper>
+            {sidebarTabs.map(tab => (
+              <Tab
+                key={tab.id}
+                isSelected={selectedTab === tab.id}
+                onClick={() => handleTabChange(tab.id)}>
+                {tab.label}
+              </Tab>
+            ))}
+          </NavBarWrapper>
+          <SidebarContentWrapper>
+            <TabContent>
+              {selectedTab && sidebarTabs.find(tab => tab.id === selectedTab)?.content}
+            </TabContent>
+            <TabButtonWrapper>
+              {selectedTab && sidebarTabs.find(tab => tab.id === selectedTab)?.tabButton1}
+              {selectedTab && sidebarTabs.find(tab => tab.id === selectedTab)?.tabButton2}
+            </TabButtonWrapper>
+          </SidebarContentWrapper>
+        </SidebarWrapper>
+      ) : (
+        <>
+          <ContentWrapper>{children}</ContentWrapper>
+          <ButtonWrapper>
+            {button2}
+            {button1}
+          </ButtonWrapper>
+        </>
+      )}
```

Comment:
Can you rethink this a little to simplify it. 
Basically, the modal has 4 areas:
1. (optional) title header
2. (optional) tab sidebar
3. content
4. button area

Since content and button area are consistent, it'd be nicer if we considered this Modal not as "either a tabular modal or a non-tabular modal". Instead, if we can break it down as areas (some of which are optional), it'll be cleaner I think.

Reply:
can we discuss this point after daily scrum